src/segmented-control/segmented-control.scss

```diff
@@ -0,0 +1,173 @@
+  // stylelint-disable-next-line primer/box-shadow
+  box-shadow: var(--primer-borderInset-thin, inset 0 0 0 $border-width) var(--color-border-default);
+}
+
+// Button -----------------------------------------
+
+.SegmentedControl-button {
+  position: relative;
+  display: inline-flex;
+  height: var(--primer-control-medium-size, 32px);
+  // stylelint-disable-next-line primer/spacing
+  padding: 0 var(--primer-control-medium-paddingInline-normal, 12px);
+  // stylelint-disable-next-line primer/typography
+  font-size: var(--primer-text-body-size-medium, $body-font-size);
+  color: var(--color-fg-default);
+  background-color: transparent;
+  // stylelint-disable-next-line primer/borders
+  border: var(--primer-borderWidth-thin, $border-width) $border-style transparent;
+  // stylelint-disable-next-line primer/borders
+  border-radius: var(--primer-borderRadius-medium, $border-radius);
+  transition: border-color, background-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  align-items: center;
+  justify-content: center;
+  gap: var(--primer-control-medium-gap, $spacer-2);
+
+  &:hover {
+    background-color: var(--color-segmented-control-hover-bg);
+    transition-duration: 120ms;
+  }
+
+  &:active {
+    background-color: var(--color-segmented-control-active-bg);
+    border-color: var(--color-segmented-control-active-border);
+    transition-duration: 60ms;
+  }
+
+  // Pressed
+
+  // TODO: Change to aria-current or aria-selected?
+  &[aria-pressed='true'] {
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    background-color: var(--color-segmented-control-pressed-bg);
+    border-color: var(--color-segmented-control-pressed-border);
+    box-shadow: var(--color-shadow-medium);
+  }
+
+  // Divider
+
+  // stylelint-disable-next-line scss/selector-no-redundant-nesting-selector
+  & + .SegmentedControl-button::before {
+    position: absolute;
+    top: 1px; // Account for border width
+    left: -1px;
+    height: var(--primer-text-body-size-large, 16px);
+    // stylelint-disable-next-line primer/spacing
+    margin-top: var(--primer-control-medium-paddingBlock, 6px);
+    content: '';
+    // stylelint-disable-next-line primer/borders
+    border-left: var(--primer-borderWidth-thin, $border-width) $border-style var(--color-border-default);
+    transition: border-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  }
+
+  // Remove dividers
+
+  &[aria-pressed='true'] + .SegmentedControl-button::before {
+    border-color: transparent;
+  }
+
+  &:hover,
+  &:active,
+  &:focus-visible {
+    &::before,
+    + .SegmentedControl-button::before {
+      border-color: transparent;
+      transition-duration: 60ms;
+    }
+  }
+}
+
+// Icon -----------------------------------------
+
+.SegmentedControl-leadingIcon {
+  color: var(--color-fg-muted);
+}
+
+// Label -----------------------------------------
+
+.SegmentedControl-label {
```

Comment:
Maybe we should rename this to `SegmentedControl-text` to not confuse with `<label>` that might appear above or `aria-label`?